### ISSUE TITLE
Mangohud seems to assume presence of wine binary

### ISSUE THREAD

**Issue by shmerl** [flightlessmango/MangoHud#895]
When using Mangohud with a game run through Wine that only has wine64 binary (non WoW64 build), periodically such errors are thrown in the log:

```
sh: 1: /opt/wine-wayland/bin/wine: not found
```

Which I assume comes from mangohud, becasue without it, it doesn't happen. It potentially assumes somewhere that Wine is always a WoW64 one.

UPDATE:

Looks like it indeed matches to loading of the layer.

**Comment by TheComputerGuy96** [comment]
Here's the (likely) problematic line: https://github.com/flightlessmango/MangoHud/blob/1a3fa6e4ff08a0eda17aa72dce4cdd4e12d84be8/src/overlay.cpp#L862

**Comment by shmerl** [comment]
Yeah, that hardcodes it to `wine` just above that:

https://github.com/flightlessmango/MangoHud/blob/1a3fa6e4ff08a0eda17aa72dce4cdd4e12d84be8/src/overlay.cpp#L858

```
findVersion << "\"" << dir << "/wine\" --version";
```

**Comment by shmerl** [comment]
It would need a check of some sort if `wine` binary exists and if not, assume it's wine64 only case.

**Comment by gort818** [comment]
I will add a check for wine64

**Comment by shmerl** [comment]
You can also use an implicit heuristic.

I.e. mango hud should be able to figure out the bitness of its own process I assume. Then you can simply check, if it's 32-bit, then always use `wine`, becasue `wine64` would be irrelevant. And if it's 64-bit, then always use wine64 becasue it should always be present in that case.

**Comment by gort818** [comment]
This patch should do it
[wine.txt](https://github.com/flightlessmango/MangoHud/files/10430661/wine.txt)

**Comment by shmerl** [comment]
That would probably work. I can try building it a bit later.

**Comment by gort818** [comment]
any luck with the patch?

**Comment by shmerl** [comment]
The patch helps to remove the error. Is there some way to display the Wine version in the hud to double check that it actually reads things correctly?

**Comment by shmerl** [comment]
OK, found it (added `wine` to `MANGOHUD_CONFIG` string). It works correctly!

**Comment by gort818** [comment]
Ok thanks fixed here f6d980696dffd4976e4fc6aa5ebd4bd1d61a4099
